pipeline.py

```diff
@@ -0,0 +1,43 @@
+import dlt, os
+from typing import Any
+from dlt.common.typing import TDataItem
+from dlt.sources.helpers import requests
+from dlt.common.plugins import CallbackPlugin, PluginConfig
+from dlt.common.configuration.specs.base_configuration import configspec
+
+
+
+@configspec
+class MyPluginConfig(PluginConfig):
+    second_name: str
+
+class MyPlugin(CallbackPlugin):
+
+    NAME: str = "my_plugin"
```

Comment:
those should be optional.
NAME should come from class name (by default)
SPEC should be None or `BaseConfiguration` by defaiult. you could also put `with_config` on an __init__ functions so you do not need any formalized way to pass SPEC. user does that or not.

Reply:
the way it is built now, the plugin name is part of the sections tuple in the config resolution. we ofc can also leave it up to the user, but as I wrote above, my hope is that these plugins can somehow be share among our users and then configured like destinations without the user having to even open the plugin code (maybe only to register the plugin)